src/core/receptor.cairo

```diff
@@ -0,0 +1,299 @@
+#[starknet::contract]
+pub mod receptor {
+    use access_control::access_control_component;
+    use core::num::traits::Zero;
+    use opus::core::roles::receptor_roles;
+    use opus::external::interfaces::ITask;
+    use opus::external::interfaces::{IEkuboOracleExtensionDispatcher, IEkuboOracleExtensionDispatcherTrait};
+    use opus::interfaces::IReceptor::IReceptor;
+    use opus::interfaces::IShrine::{IShrineDispatcher, IShrineDispatcherTrait};
+    use opus::types::QuoteTokenInfo;
+    use opus::utils::math::{median_of_three, pow, scale_x128_to_wad};
+    use starknet::{ContractAddress, get_block_timestamp};
+    use wadray::Wad;
+
+    //
+    // Components
+    //
+
+    component!(path: access_control_component, storage: access_control, event: AccessControlEvent);
+
+    #[abi(embed_v0)]
+    impl AccessControlPublic = access_control_component::AccessControl<ContractState>;
+    impl AccessControlHelpers = access_control_component::AccessControlHelpers<ContractState>;
+
+    //
+    // Constants
+    //
+
+    const LOOP_START: u32 = 1;
+    pub const NUM_QUOTE_TOKENS: u32 = 3;
+    pub const LOWER_UPDATE_FREQUENCY_BOUND: u64 = 15; // seconds (approx. Starknet block prod goal)
+    pub const UPPER_UPDATE_FREQUENCY_BOUND: u64 = 4 * 60 * 60; // 4 hours * 60 minutes * 60 seconds
+
```

Comment:
No strong opinion, but maybe we should have a min duration bound as well? WDYT?

Reply:
I think it's sufficient to have only a lower bound on the duration value, so yeah, 1 min as a sanity check sounds good.